src/Xamarin.Android.Build.Tasks/Utilities/TypeMapGenerator.cs

```diff
@@ -87,7 +87,22 @@ public TypeMapGenerator (Action<string> logger, string[] supportedAbis)
 			typemapIndexMagicString = outputEncoding.GetBytes (TypeMapIndexMagicString);
 		}
 
-		public bool Generate (List<TypeDefinition> javaTypes, string outputDirectory, bool generateNativeAssembly)
+		void UpdateApplicationConfig (TypeDefinition javaType, ApplicationConfigTaskState appConfState)
+		{
+			if (appConfState.JniAddNativeMethodRegistrationAttributePresent)
+				return;
+			if (!javaType.HasCustomAttributes)
+				return;
+
+			foreach (CustomAttribute ca in javaType.CustomAttributes) {
+				if (!appConfState.JniAddNativeMethodRegistrationAttributePresent && String.Compare ("JniAddNativeMethodRegistrationAttribute", ca.AttributeType.Name, StringComparison.Ordinal) == 0) {
```

Comment:
You can actually use the static `string.Equals` and don't have to use `== 0`:
```suggestion
				if (!appConfState.JniAddNativeMethodRegistrationAttributePresent && String.Equals ("JniAddNativeMethodRegistrationAttribute", ca.AttributeType.Name, StringComparison.Ordinal)) {
```

Reply:
It seems (from looking at the corefx source) `.Compare` might be slightly faster than `String.Equals` and I don't mind the extra verbosity of the code

Reply:
Link?

Reply:
Actually, they are equal...

Reply:
It does actually seem faster:
```
// * Summary *

BenchmarkDotNet=v0.12.0, OS=Windows 10.0.18362
Intel Core i9-9900K CPU 3.60GHz (Coffee Lake), 1 CPU, 16 logical and 8 physical cores
  [Host]     : .NET Framework 4.8 (4.8.4121.0), X86 LegacyJIT
  DefaultJob : .NET Framework 4.8 (4.8.4121.0), X86 LegacyJIT


|  Method |      Mean |     Error |    StdDev | Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------- |----------:|----------:|----------:|------:|------:|------:|----------:|
| Compare | 0.9241 ns | 0.0078 ns | 0.0073 ns |     - |     - |     - |         - |
|  Equals | 1.4317 ns | 0.0122 ns | 0.0102 ns |     - |     - |     - |         - |
```
```csharp
[Benchmark]
public void Compare ()
{
	if (string.Compare ("ABCD", "ABCD", StringComparison.Ordinal) == 0) {
		int x; //Just need some instruction
	}
}

[Benchmark]
public void Equals ()
{
	if (string.Equals ("ABCD", "ABCD", StringComparison.Ordinal)) {
		int x; //Just need some instruction
	}
}
```

Reply:
https://github.com/mono/corefx/blob/7c9e2158925339a5a96ff8d09d21c600a0b27e5a/src/Common/src/CoreLib/System/String.Comparison.cs#L268 vs https://github.com/mono/corefx/blob/7c9e2158925339a5a96ff8d09d21c600a0b27e5a/src/Common/src/CoreLib/System/String.Comparison.cs#L727

Reply:
Changed it back to `Compare` :)